include/ck_tile/ops/grouped_convolution/kernel/grouped_convolution_forward_kernel.hpp

```diff
@@ -844,13 +884,141 @@ struct GroupedConvolutionForwardKernel
             static_cast<long_index_t>(batch_offset) *
             static_cast<long_index_t>(kargs.output_batch_stride);
 
-        // Adjust pointers: combine group offset and batch offset
-        const InDataType* a_ptr =
-            static_cast<const InDataType*>(kargs.in_ptr) + group_offset_a + input_batch_offset;
+        // Adjust pointers with formula: base + group_offset + batch_offset + spatial_offset
+        // This ensures spatial offset is applied per-batch, not globally (standard behavior)
+        const InDataType* base_a_ptr =
+            static_cast<const InDataType*>(kargs.in_ptr) + group_offset_a + input_batch_offset +
+            kargs.spatial_offset_in; // Add spatial offset from split-image
         const WeiDataType* b_ptr = static_cast<const WeiDataType*>(kargs.wei_ptr) +
                                    group_offset_b; // No batch offset for weights!
-        OutDataType* c_ptr =
-            static_cast<OutDataType*>(kargs.out_ptr) + group_offset_c + output_batch_offset;
+        OutDataType* base_c_ptr = static_cast<OutDataType*>(kargs.out_ptr) + group_offset_c +
+                                  output_batch_offset +
+                                  kargs.spatial_offset_out; // Add spatial offset from split-image
+
+        // =====================================================================
+        // Split-image: Map local block to global tile index (if enabled)
+        // =====================================================================
+        const InDataType* a_ptr = base_a_ptr;
+        OutDataType* c_ptr      = base_c_ptr;
+        index_t i_m             = 0;
+        index_t i_n             = 0;
+
+        // Pre-calculate block_id (used in both split-image and non-split paths)
+        const index_t block_id = static_cast<index_t>(blockIdX);
+
+        if(kargs.num_spatial_pieces > 1)
+        {
+            // Helper: Find which piece owns this block
+            auto find_piece_id = [&]() -> index_t {
+                for(index_t i = 0; i < kargs.num_spatial_pieces; i++)
```

Comment:
you can use golden division like in include/ck/tensor_operation/gpu/device/impl/device_grouped_conv_fwd_multiple_d_xdl_large_tensor_cshuffle.hpp